src/platform/cc13x2_26x2/BLEManagerImpl.cpp

```diff
@@ -0,0 +1,1885 @@
+        HandleSubscribeReceived(event->CHIPoBLESubscribe.ConId, &CHIP_BLE_SVC_ID, &chipUUID_CHIPoBLEChar_TX);
+
+        connEstEvent.Type = DeviceEventType::kCHIPoBLEConnectionEstablished;
+
+        PlatformMgr().PostEvent(&connEstEvent);
+    }
+    break;
+
+    case DeviceEventType::kCHIPoBLEUnsubscribe: {
+        BLEMGR_LOG("BLEMGR: OnPlatformEvent, kCHIPoBLEUnsubscribe");
+        HandleUnsubscribeReceived(event->CHIPoBLEUnsubscribe.ConId, &CHIP_BLE_SVC_ID, &chipUUID_CHIPoBLEChar_TX);
+    }
+    break;
+
+    case DeviceEventType::kCHIPoBLEWriteReceived: {
+        BLEMGR_LOG("BLEMGR: OnPlatformEvent, kCHIPoBLEWriteReceived");
+        HandleWriteReceived(event->CHIPoBLEWriteReceived.ConId, &CHIP_BLE_SVC_ID, &chipUUID_CHIPoBLEChar_RX,
+                            PacketBufferHandle::Adopt(event->CHIPoBLEWriteReceived.Data));
+    }
+    break;
+
+    case DeviceEventType::kCHIPoBLEIndicateConfirm:
+        HandleIndicationConfirmation(event->CHIPoBLEIndicateConfirm.ConId, &CHIP_BLE_SVC_ID, &chipUUID_CHIPoBLEChar_TX);
+        break;
+
+    case DeviceEventType::kCHIPoBLEConnectionError: {
+        BLEMGR_LOG("BLEMGR: OnPlatformEvent, kCHIPoBLEConnectionError");
+        HandleConnectionError(event->CHIPoBLEConnectionError.ConId, event->CHIPoBLEConnectionError.Reason);
+    }
+    break;
+
+    default:
+        break;
+    }
+}
+
+// ===== Members that implement virtual methods on BlePlatformDelegate.
+bool BLEManagerImpl::CloseConnection(BLE_CONNECTION_OBJECT conId)
+{
+    void * pMsg = (void *) ICall_malloc(sizeof(void *));
+    pMsg        = (void *) conId;
+
+    EnqueueEvtHdrMsg(BLEManagerIMPL_CHIPOBLE_CLOSE_CONN_EVT, (void *) pMsg);
+
+    return CHIP_NO_ERROR;
+}
+
+uint16_t BLEManagerImpl::GetMTU(BLE_CONNECTION_OBJECT conId) const
+{
+    uint8_t index;
+    uint16_t mtu = 0;
+
+    index = GetBLEConnIndex(*((uint32_t *) conId));
+
+    if (index != MAX_NUM_BLE_CONNS)
+    {
+        mtu = sInstance.connList[index].mtu;
+        /* Prior to MTU update event, MTU is determined by the below formula */
+        if (mtu == 0)
+        {
+            mtu = MAX_PDU_SIZE - 4;
+        }
+    }
+
+    return mtu;
+}
+
+// ===== Members that implement virtual methods on BleApplicationDelegate.
+
+void BLEManagerImpl::NotifyChipConnectionClosed(BLE_CONNECTION_OBJECT conId)
+{
+    // Unused
+}
+
+bool BLEManagerImpl::SendIndication(BLE_CONNECTION_OBJECT conId, const ChipBleUUID * svcId, const ChipBleUUID * charId,
+                                    PacketBufferHandle data)
+{
+    BLEMGR_LOG("BLEMGR: BLE SendIndication ");
+
+    // Allocate buffers to send to BLE app task
+    uint8_t dataLen = static_cast<uint8_t>(data->DataLength());
+    CHIPoBLEIndEvt_t * pMsg;
+    uint8_t * pBuf;
+
+    pMsg = (CHIPoBLEIndEvt_t *) ICall_malloc(sizeof(CHIPoBLEIndEvt_t));
+    if (NULL == pMsg)
+    {
+        return false;
+    }
+
+    pBuf = (uint8_t *) ICall_malloc(dataLen);
+    if (NULL == pBuf)
+    {
+        ICall_free((void *) pBuf);
```

Comment:
is this suppose to be `ICall_free((void *) pMsg);`

Reply:
good catch, yes it is.

Reply:
Thanks for the quick response.  Waiting for all of the checks to complete, and I'll merge this in.